### PR TITLE
fix(core/dropdown): adjusted height for dropdown (siemens#1671)

### PR DESCRIPTION
**Pull request by varun-srinivasa**
## 💡 What is the current behavior?

GitHub Issue Number: #1671 

## 🆕 What is the new behavior?

- Change max-height of dropdown from 50vh to calc(50vh - 1px)
- Added visual regression test cases

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [X] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📄 Documentation was reviewed/updated (`pnpm run docs`)
- [ ] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [X] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [ ] 🧐 Static code analysis passes (`pnpm lint`)
- [X] 🏗️ Successful compilation (`pnpm build`, changes pushed)

### CODE REVIEW

**Review by danielleroux — COMMENTED** [review]
Thanks for your contribution, looks like your screenshots for testing are generated with windows. You have to generate it via a linux based system: https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing

File: packages/core/src/components/dropdown/dropdown.scss [hunk 1]
```diff
@@ -33,7 +33,7 @@
 }
 
 :host(.overflow) {
-  max-height: 50vh;
+  max-height: calc(50vh - 50px);
```
Comment (danielleroux) [inline packages/core/src/components/dropdown/dropdown.scss:36]: Maybe we can use here a variable which makes it looks less like a magic number `50px`

File: packages/core/src/tests/dropdown/centered-overflow/index.html
Comment (danielleroux) [inline packages/core/src/tests/dropdown/centered-overflow/index.html:1]: Just thought myself is a visual-regression test here really necessary? Should be also possible via a simple component test which makes snapshot creation not necessary.

File: packages/core/src/components/dropdown/dropdown.scss (hunk 1 above)
Comment (nuke-ellington) [inline packages/core/src/components/dropdown/dropdown.scss:36]: `Rem` should be used over `px`.

File: packages/core/src/tests/dropdown/centered-overflow/index.html
Comment (nuke-ellington) [reply packages/core/src/tests/dropdown/centered-overflow/index.html:1]: Additional/edge cases (e.g. dropdown opened via trigger element, dropdown opening above trigger) are missing.

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

**Comment by danielleroux** [comment]
@varun-srinivasa did you consider to use a component test instead of an visual-regression tests?

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Visual regression tests of other components that utilise the ix-dropdown fail with these changes. Please review them and decide if these changes need to be fixed or the snapshots need to be udpted.

File: packages/core/src/components/dropdown/dropdown.scss [hunk 2]
```diff
@@ -32,8 +32,10 @@
   }
 }
 
+$dropdown-offset: 3.125rem;
```
Comment (nuke-ellington) [inline packages/core/src/components/dropdown/dropdown.scss:35]: Why exactly this value, when the ix-select e.g. has a height of 32px?
IMO we should make the offset based on the dropdown's trigger/anchor element.

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/components/dropdown/dropdown.scss (hunk 2 above)
Comment (nuke-ellington) [inline packages/core/src/components/dropdown/dropdown.scss:35]: This solution seems fine after all.
Can we change the offset to 3rem to be more inline with other spacings?
This would require updating the visual test snapshots.

```suggestion
$dropdown-offset: 3rem;
```

Comment (varun-srinivasa) [reply packages/core/src/components/dropdown/dropdown.scss:35]: Hi, yes I will update it to 3, test and update in the PR

File: packages/core/src/components/dropdown/test/dropdown.ct.ts [hunk 3]
```diff
@@ -742,3 +742,84 @@ test('Dropdown works in floating-ui', async ({ mount, page }) => {
...
+      ).join('')}
+    </ix-dropdown>`);
+
+  await page.locator('#trigger').click();
+
+  const dropdown = page.locator('ix-dropdown');
+  await expect(dropdown).toBeVisible();
+
+  await page.evaluate(() => {
+    const dropdown = document.querySelector('ix-dropdown');
```
Comment (nuke-ellington) [inline packages/core/src/components/dropdown/test/dropdown.ct.ts:762]: there is already a locator for the dropdown above

File: packages/core/src/components/dropdown/test/dropdown.ct.ts [hunk 4]
```diff
@@ -742,3 +742,84 @@ test('Dropdown works in floating-ui', async ({ mount, page }) => {
...
+    const lastItem = dropdown?.querySelector('ix-dropdown-item:last-child');
```
Comment (nuke-ellington) [inline packages/core/src/components/dropdown/test/dropdown.ct.ts:763]: Use locator instead of querySelector

File: packages/core/src/components/dropdown/test/dropdown.ct.ts [hunk 5]
```diff
@@ -742,3 +742,84 @@ test('Dropdown works in floating-ui', async ({ mount, page }) => {
...
+    if (lastItem) {
+      lastItem.scrollIntoView();
+    }
+  });
+
+  const lastItem = page.locator('ix-dropdown-item').last();
```
Comment (nuke-ellington) [inline packages/core/src/components/dropdown/test/dropdown.ct.ts:769]: reuse locator from above comment

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Please remove all console logs from the test files.

File: packages/core/src/components/select/test/select.ct.ts [hunk 6]
```diff
@@ -770,3 +770,58 @@ test('should preserve spaces within input and show add icon', async ({
...
+  await page.locator('[data-select-dropdown]').click();
+
+  const dropdown = select.locator('ix-dropdown');
+  await expect(dropdown).toBeVisible();
+
+  const lastItem = page.locator('ix-select-item').last();
+  await lastItem.click();
+
+  const inputValue = await select.locator('input').inputValue();
+  expect(inputValue).toEqual('Item 20');
```
Comment (nuke-ellington) [inline packages/core/src/components/select/test/select.ct.ts:802]:
```suggestion
  await lastItem.evaluate((item) => {
    item.scrollIntoView();
  });
  await expect(lastItem.toBeVisible());
```

File: packages/core/src/components/select/test/select.ct.ts [hunk 7]
```diff
@@ -770,3 +770,58 @@ test('should preserve spaces within input and show add icon', async ({
   await expect(addItem).toHaveCount(1);
   await expect(input).toHaveValue(itemText);
 });
+
+test('select and display the last element in the list', async ({
```
Comment (nuke-ellington) [inline packages/core/src/components/select/test/select.ct.ts:774]:
```suggestion
test('last select item can be accessed via scrolling, async ({
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

File: packages/core/src/tests/dropdown/dropdown.e2e.ts [hunk 8]
```diff
@@ -78,4 +78,18 @@ regressionTest.describe('dropdown', () => {
...
+    await page.goto('dropdown/centered-overflow');
+
+    const menuHandle = await page.waitForSelector('.dropdown-menu.show');
+
+    page.evaluate((menuElement) => {
+      menuElement.scrollTop = 9999;
+      menuElement.classList.add('__SCROLLED__');
+    }, menuHandle);
+
+    await page.waitForSelector('.dropdown-menu.show.__SCROLLED__');
```
Comment (nuke-ellington) [inline packages/core/src/tests/dropdown/dropdown.e2e.ts:92]: Something like this should work:
```suggestion
    const lastItem = await page.locator('dropdown-item').last();
    await lastItem.scrollIntoViewIfNeeded();
```

File: packages/core/src/tests/dropdown/centered-overflow/index.html [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    <meta charset="utf-8" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0"
+    />
+    <title>Stencil Component Starter</title>
+  </head>
+  <body style="width: 100vw; height: 100vh;">
+    
+    <div style="display: block; position: relative; width: 100rem; margin-top: 53vh; height: 90%">
```
Comment (nuke-ellington) [inline packages/core/src/tests/dropdown/centered-overflow/index.html:19]:
```suggestion
    <div style="display: block; position: relative; width: 100rem; margin-top: calc(50vh - 1px)">
```

File: packages/core/src/tests/select/centered-overflow/index.html [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0"
+    />
+    <title>Stencil Component Starter</title>
+  </head>
+  
+    <body style="width: 100vw; height: 100vh; padding-top: 52vh;">
```
Comment (nuke-ellington) [inline packages/core/src/tests/select/centered-overflow/index.html:18]:
```suggestion
    <body style="width: 100vw; height: 100vh; padding-top: calc(50vh - 1px);">
```

File: packages/core/src/tests/select/select.e2e.ts [hunk 11]
```diff
@@ -125,6 +125,20 @@ regressionTest.describe('select', () => {
...
+    await page.goto('select/centered-overflow');
+    await page.locator('ix-select').locator('[data-select-dropdown]').click();
+    const menuHandle = await page.waitForSelector('.dropdown-menu.show');
+
+    await page.evaluate((menuElement) => {
+      menuElement.scrollTop = 9999;
+      menuElement.classList.add('__SCROLLED__');
+    }, menuHandle);
+
+    await page.waitForSelector('.dropdown-menu.show.__SCROLLED__');
```
Comment (nuke-ellington) [inline packages/core/src/tests/select/select.e2e.ts:138]: Same as above
```suggestion
    await page.evaluate((menuElement) => {
      menuElement.scrollTop = 9999;
      menuElement.classList.add('__SCROLLED__');
    }, menuHandle);

    await page.waitForSelector('.dropdown-menu.show.__SCROLLED__');
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]

**Review by nuke-ellington — APPROVED** [review]
